### PR TITLE
Implement new initiate validator exit function

### PR DESCRIPTION
**Pull request by terencechain**
Part of #2307 

Updated:
- Implemented [InitiateValidatorExit](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#initiate_validator_exit)
- Replaced old `InitiateValidatorExit`
- Fixed existing tests and added tests for `InitiateValidatorExit `

### CODE REVIEW

File: beacon-chain/core/validators/validator.go [hunk 1]
```diff
@@ -179,11 +179,70 @@ func ActivateValidator(state *pb.BeaconState, idx uint64, genesis bool) (*pb.Bea
...
+	v := state.ValidatorRegistry[idx]
+
+	// Return if validator already initiated exit.
+	// According to the spec, this is not an assert condition and
+	// shouldn't fail beacon block state transition.
+	if v.ExitEpoch != params.BeaconConfig().FarFutureEpoch {
+		return state
+	}
+
+	// Aggregate all the exit epochs from the last.
```
Comment (prestonvanloon) [inline beacon-chain/core/validators/validator.go]: From the last what?

File: beacon-chain/core/validators/validator.go [hunk 2]
```diff
@@ -179,11 +179,70 @@ func ActivateValidator(state *pb.BeaconState, idx uint64, genesis bool) (*pb.Bea
...
+	var exitedEpochs []uint64
```
Comment (prestonvanloon) [inline beacon-chain/core/validators/validator.go]: It looks like you are looking only for the highest epoch, no need to store all of them in an array and iterate over them to find the highest epoch.

File: beacon-chain/core/validators/validator.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/core/validators/validator.go]: removed

File: beacon-chain/core/validators/validator.go (hunk 2 above)
Comment (terencechain) [reply beacon-chain/core/validators/validator.go]: great feedback, this simplified it by a lot

**Pull request merged**